### DISCUSSION TITLE
Unit testing RouteClaimsRequirement

### DISCUSSION THREAD

**Discussion by aleroca96** [ThreeMammals/Ocelot#2068] [Q&A] [answered]
Hi

I am looking to write unit tests for specific routes that have RouteClaimsRequirement.

I am struggling to find the services to mock in order to avoid calling the end services while still being able to mock the claims logic.

Apologies if I am missing the solution to this issue on the docs.

Thanks! :)

**Comment by raman-m** [comment, marked as answer]
- [Search for RouteClaimsRequirement](https://ocelot.readthedocs.io/en/latest/search.html?q=RouteClaimsRequirement) in docs
- [RouteClaimsRequirement in C#](https://github.com/search?q=repo%3AThreeMammals%2FOcelot+RouteClaimsRequirement+language%3AC%23&type=code&l=C%23) 

I believe you need to **mock** injected services of the [ClaimsAuthorizer](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/Authorization/ClaimsAuthorizer.cs) class 👇 
https://github.com/ThreeMammals/Ocelot/blob/6e9a975fae3b26229c3f0b3c1a475a7b05e95fee/src/Ocelot/Authorization/ClaimsAuthorizer.cs#L10-L15

So, you have to mock `IClaimsParser` interface object, and/or `IClaimsAuthorizer` object.
But exact mocking depends on the unit test and testing scenario.

Anything else?

**Reply by aleroca96** [reply]
brilliant, thanks a lot!

**Reply by raman-m** [reply]
You are welcome!
